app/services/browser_support.rb

```diff
@@ -0,0 +1,62 @@
+class BrowserSupport
+  @cache = LruRedux::Cache.new(1_000)
+
+  BROWSERSLIST_TO_BROWSER_MAP = {
+    and_chr: :chrome?,
+    and_uc: :uc_browser?,
+    chrome: :chrome?,
+    edge: :edge?,
+    firefox: :firefox?,
+    ios_saf: :safari?,
+    op_mini: :opera_mini?,
+    opera: :opera?,
+    safari: :safari?,
+    samsung: :samsung_browser?,
+  }.transform_values { |method| Browser::Base.instance_method(method) }.
+    with_indifferent_access.
+    freeze
+
+  class << self
+    attr_reader :cache
+
+    def supported?(user_agent)
+      return false if user_agent.nil?
+      return true if browser_support_config.nil?
+
+      cache.getset(user_agent) do
+        matchers.any? { |matcher| matcher.call(BrowserCache.parse(user_agent)) }
+      end
+    end
+
+    private
+
+    def matchers
+      @matchers ||= browser_support_config.flat_map do |config_entry|
+        key, version = config_entry.split(' ', 2)
+        browser_matcher = BROWSERSLIST_TO_BROWSER_MAP[key]
+        next [] if !browser_matcher
+
+        low_version, = version.split('-', 2)
```

Comment:
ooo I didn't realize ruby let us not put a variable name on the right side of a comma, neat

Reply:
> ooo I didn't realize ruby let us not put a variable name on the right side of a comma, neat

Yeah, though on second glance, maybe this is easy to miss, and naming the second part could make the splitting more clear, as well as communicating the other portion of the split result.

```suggestion
        low_version, _high_version = version.split('-', 2)
```

Reply:
🤷 I could go either way

Reply:
> > ooo I didn't realize ruby let us not put a variable name on the right side of a comma, neat
> 
> Yeah, though on second glance, maybe this is easy to overlook, and naming the second part could make the splitting more clear, as well as communicating the other portion of the split result.

Named second part of split value in 36c04000b.